### ISSUE TITLE
Problem in label2vec while simple net training

### ISSUE THREAD

**Issue by MaciejKolodziej** [tiny-dnn/tiny-dnn#836]
Hello
I'm trying to train following network:
`network<sequential> net;

	net << layers::dense(400, 200) << activation::softmax()
		<< layers::dense(200, 32) << activation::softmax()
		<< layers::dense(32, 1) << activation::sigmoid();

	assert(net.in_data_size() == 400);
	assert(net.out_data_size() == 1);

	std::vector<vec_t> inputsVecT = toVecT(inputs);
	std::vector<label_t> outputsVecT = toLabelT(outputs);

	auto splitFactor = 0.65f;
	std::vector<vec_t> train_x = getVectorSubset(inputsVecT, 0, (int)(inputs.size() * splitFactor));
	std::vector<label_t> train_y = getVectorSubset(outputsVecT, 0, (int)(inputs.size() * splitFactor));

	std::vector<vec_t> test_x = getVectorSubset(inputsVecT, (int)(inputs.size() * splitFactor), inputs.size() - 1);
	std::vector<label_t> test_y = getVectorSubset(outputsVecT, (int)(inputs.size() * splitFactor), inputs.size() - 1);

	RMSprop optimizer;
	const clock_t clockEpochStart = clock();

	auto onBatchEnd = [&]() {
		std::cout << "\n Epoch time: " << (clock() - clockEpochStart) / (CLOCKS_PER_SEC / 1000) << std::endl;
	};
	auto onEpochEnd = [&]() {
		net.test(test_x, test_y).print_detail(std::cout);
	};

	if (net.train<cross_entropy, RMSprop>(optimizer, train_x, train_y, 50, 150, onBatchEnd, onEpochEnd, false, 4, std::vector<vec_t>())) {
		net.test(test_x, test_y).print_detail(std::cout);
	}
	else {
		std::cout << "\n training failed!\n";
	}`
Unfortunately I'm getting into failed assert in 'nodes.h' at:
`  void label2vec(const label_t *t, size_t num, std::vector<vec_t> &vec) const {
    size_t outdim = out_data_size();

    vec.reserve(num);
    for (size_t i = 0; i < num; i++) {
      assert(t[i] < outdim);
      vec.emplace_back(outdim, target_value_min());
      vec.back()[t[i]] = target_value_max();
    }
  }`
t[i] === 1
outdim === 1
Why can't I have values for output of equal size or bigger as output dimension?
I'm just learning ML and  tiny-dnn, but this seems quite unexpected as exactly same model worked great in keras.

**Comment by Sophia0130** [comment]
I have the same problem.
Do you know ho to solve it?
